renderdoc/driver/dxgi/dxgi_common.cpp

```diff
@@ -1449,6 +1449,8 @@ DXGI_FORMAT MakeDXGIFormat(ResourceFormat fmt)
   {
     if(fmt.compByteWidth == 4)
       ret = DXGI_FORMAT_R32_TYPELESS;
+    else if(fmt.compByteWidth == 3)
+      ret = DXGI_FORMAT_R24_UNORM_X8_TYPELESS;
```

Comment:
Right, if the incoming format is actually D24S8 then everything should work fine already. My concern is that the incoming format is D24, but here we are setting it to D24S8, so we're pretending it has more data than it really does. I'm not sure if that mismatch might cause problems.

I can't remember if when data is transferred for D24S8 it's stored unpacked: 4 bytes for the depth (3 are used), and 4 bytes for stencil (1 is used). If that's what happens, then when setting the data for this format it will expect more data than it actually gets, since we'll only have 4 bytes for depth with 1 unused, and no stencil.

It would be good to be sure about what the data looks like, if it's packed together in 4 bytes like it is on the GPU, then these formats are equivalent as the extra stencil doesn't really change anything with the data layout.